### PR TITLE
[alerting/metrics] added gauge/alert for time since last normalize

### PR DESCRIPTION
**Pull request by heavycrystal**

### CODE REVIEW

File: flow/activities/flowable.go [hunk 1]
```diff
@@ -767,6 +767,17 @@ func (a *FlowableActivity) RecordSlotSizes(ctx context.Context) error {
 					return
 				}
 				slotMetricGauges.OpenReplicationConnectionsGauge = openReplicationConnectionsGauge
+
+				intervalSinceLastNormalizeGauge, err := otel_metrics.GetOrInitFloat64SyncGauge(a.OtelManager.Meter,
+					a.OtelManager.Float64GaugesCache,
+					peerdb_gauges.IntervalSinceLastNormalizeGaugeName,
```
Comment (iamKunalGupta) [inline flow/activities/flowable.go:773]: Pull from main and use the `Build` method

Comment (iamKunalGupta) [reply flow/activities/flowable.go:773]: I was thinking we should record metrics a bit more often

File: flow/connectors/postgres/postgres.go [hunk 2]
```diff
@@ -1207,6 +1207,14 @@ func (c *PostgresConnector) HandleSlotInfo(
 		attribute.String(peerdb_gauges.PeerNameKey, alertKeys.PeerName),
 		attribute.String(peerdb_gauges.DeploymentUidKey, peerdbenv.PeerDBDeploymentUID())))
 
+	intervalSinceLastNormalize := alerter.AlertIfTooLongSinceLastNormalize(ctx, alertKeys)
```
Comment (iamKunalGupta) [inline flow/connectors/postgres/postgres.go:1210]: If alerting is disabled, we may not get the metric value

File: flow/alerting/alerting.go [hunk 3]
```diff
@@ -244,6 +248,60 @@ func (a *Alerter) AlertIfOpenConnections(ctx context.Context, alertKeys *AlertKe
 	}
 }
 
+func (a *Alerter) AlertIfTooLongSinceLastNormalize(ctx context.Context, alertKeys *AlertKeys) *time.Duration {
+	intervalSinceLastNormalizeThreshold, err := peerdbenv.PeerDBIntervalSinceLastNormalizeThresholdMinutes(ctx, nil)
+	if err != nil {
+		logger.LoggerFromCtx(ctx).Warn("failed to get open connections alert threshold from catalog", slog.Any("error", err))
```
Comment (iamKunalGupta) [inline flow/alerting/alerting.go:254]: nit: fix this

**Review by iamKunalGupta — CHANGES_REQUESTED** [review]

File: flow/connectors/postgres/postgres.go [hunk 4]
```diff
@@ -1207,12 +1207,23 @@ func (c *PostgresConnector) HandleSlotInfo(
...
 		attribute.String(peerdb_gauges.DeploymentUidKey, peerdbenv.PeerDBDeploymentUID())))
 
-	intervalSinceLastNormalize := alerter.AlertIfTooLongSinceLastNormalize(ctx, alertKeys)
+	var intervalSinceLastNormalize *time.Duration
+	err = alerter.CatalogPool.QueryRow(ctx, "SELECT now()-max(end_time) FROM peerdb_stats.cdc_batches WHERE flow_name=$1",
+		alertKeys.FlowName).Scan(&intervalSinceLastNormalize)
+	if err != nil {
+		logger.Warn("failed to get interval since last normalize", slog.Any("error", err))
+	}
+	// what if the first normalize errors out/hangs?
```
Comment (iamKunalGupta) [inline flow/connectors/postgres/postgres.go:1216]: In that case can we check first sync end time?

File: flow/connectors/postgres/postgres.go [hunk 5]
```diff
@@ -1207,12 +1207,23 @@ func (c *PostgresConnector) HandleSlotInfo(
 		attribute.String(peerdb_gauges.PeerNameKey, alertKeys.PeerName),
 		attribute.String(peerdb_gauges.DeploymentUidKey, peerdbenv.PeerDBDeploymentUID())))
 
-	intervalSinceLastNormalize := alerter.AlertIfTooLongSinceLastNormalize(ctx, alertKeys)
+	var intervalSinceLastNormalize *time.Duration
+	err = alerter.CatalogPool.QueryRow(ctx, "SELECT now()-max(end_time) FROM peerdb_stats.cdc_batches WHERE flow_name=$1",
+		alertKeys.FlowName).Scan(&intervalSinceLastNormalize)
+	if err != nil {
+		logger.Warn("failed to get interval since last normalize", slog.Any("error", err))
```
Comment (iamKunalGupta) [inline flow/connectors/postgres/postgres.go:1214]: This should probably be an error log

File: flow/connectors/postgres/postgres.go (hunk 4 above)
Comment (heavycrystal) [reply flow/connectors/postgres/postgres.go:1216]: that isn't stored in this table, or any table afaik

File: flow/connectors/postgres/postgres.go (hunk 5 above)
Comment (heavycrystal) [reply flow/connectors/postgres/postgres.go:1214]: surrounding/alerting code generally sticks to warnings

**Review by iamKunalGupta — APPROVED** [review]

**Review by iskakaushik — APPROVED** [review]

**Pull request merged**